src/gluonts/time_feature/lag.py

```diff
@@ -51,6 +51,28 @@ def get_granularity(freq_str: str) -> Tuple[int, str]:
     return multiple, granularity
 
 
+# A dictionary mapping granularity to the period length of the longest season
+# one can expect given the granularity of the time series.
+# This is similar to the frequency value in the R forecast package:
+# https://stats.stackexchange.com/questions/120806/frequency-value-for-seconds-minutes-intervals-data-in-r
+# This is useful for setting default values for past/context length for models
+# that do not do data augmentation and uses a single training example per time series in the dataset.
+FREQ_LONGEST_PERIOD_DICT = {
+    "M": 12,  # yearly seasonality
+    "W": 52,  # yearly seasonality
+    "D": 365,  # yearly seasonality
+    "B": 365,  # yearly seasonality
+    "H": 168,  # weekly seasonality
+    "T": 1440,  # daily seasonality
+    "min": 1440,  # daily seasonality
+}
+
+
+def longest_period_from_frequency_str(freq_str: str) -> int:
+    multiple, granularity = get_granularity(freq_str)
+    return FREQ_LONGEST_PERIOD_DICT[granularity] // multiple
+
```

Comment:
Also: similar code to the one here appears in the `evaluation` package, [see here](https://github.com/awslabs/gluon-ts/blob/d6bc5468c28fa665b2d3d0e57e1a62221231f481/src/gluonts/evaluation/_base.py#L30-L54).

These two are of course different: in the `evaluation` package, for example, a seasonality of length 1 is used for `"D"` or `"W"`, and a seasonality of length 24 is used for `"H"`.

I'm not sure why that is the case, but I think that having similar pieces of code lying around can create confusion: I suggest keeping these definition here local to the `model.deepstate` package, since this is only used there. Then we will see about consolidating these.